frontend/src/concepts/pipelines/content/pipelinesDetails/pipelineRun/runLogs/usePodStepsStates.ts

```diff
@@ -0,0 +1,32 @@
+import * as React from 'react';
+import { PodContainer } from '~/types';
+import { usePipelinesAPI } from '~/concepts/pipelines/context';
+import { useDeepCompareMemoize } from '~/utilities/useDeepCompareMemoize';
+import { getPodStepsStates } from '~/concepts/k8s/pods/utils';
+
+const usePodStepsStates = (
+  podContainers: PodContainer[],
+  podName: string,
+): {
+  stepName: string;
+  state: string;
+}[] => {
+  const { namespace } = usePipelinesAPI();
+  const podSteps = useDeepCompareMemoize(podContainers);
+  const [allpodStepsState, setAllPodStepsState] = React.useState<
+    { stepName: string; state: string }[]
+  >([]);
+
+  React.useEffect(() => {
+    getPodStepsStates(podSteps, namespace, podName)
+      .then((result) => {
+        setAllPodStepsState(result);
+      })
+      .catch((error) => {
+        throw new Error(error);
+      });
```

Comment:
Why did you catch the error and just throw it? If anything wrong when fetching the logs, what are you going to do? You could handle it here, or you could use a state here and return it to the component that uses the hook(which is `LogsTabForPodName` here) to deal with it.

Reply:
I'm using a `Promise.reject` now. Is that fine?

Reply:
Thinking whether I should keep this `catch` in the first place. I don't find it in the other custom hooks used in `LogsTab`

Reply:
You could remove it, but let's see what will happen. Now the error is not handled here, so it will be caught in `LogsTabForPodName` component. The error is not handled there, so it could crash the page.